flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,278 @@
+                            Row.of("f", Vectors.dense(2.2, 3.2)),
+                            Row.of("e", Vectors.dense(2.4, 3.2)),
+                            Row.of("e", Vectors.dense(2.5, 3.2)),
+                            Row.of("e", Vectors.dense(2.5, 3.2)),
+                            Row.of("f", Vectors.dense(2.1, 3.1))));
+
+    List<Row> testArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of("e", Vectors.dense(4.0, 4.1)),
+                            Row.of("m", Vectors.dense(300, 42))));
+    private Table testData;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.STRING())
+                        .column("f1", DataTypes.of(DenseVector.class))
+                        .build();
+
+        DataStream<Row> dataStream = env.fromCollection(trainArray);
+        trainData = tEnv.fromDataStream(dataStream, schema).as(LABEL_COL + "," + VEC_COL);
+
+        DataStream<Row> predDataStream = env.fromCollection(testArray);
+        testData = tEnv.fromDataStream(predDataStream, schema).as(LABEL_COL + "," + VEC_COL);
+    }
+
+    // Executes the graph and returns a list which has true label and predict label.
+    private static List<Tuple2<String, String>> executeAndCollect(Table output) throws Exception {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) output).getTableEnvironment();
+
+        DataStream<Tuple2<String, String>> stream =
+                tEnv.toDataStream(output)
+                        .map(
+                                new MapFunction<Row, Tuple2<String, String>>() {
+                                    @Override
+                                    public Tuple2<String, String> map(Row row) {
+                                        return Tuple2.of(
+                                                (String) row.getField(LABEL_COL),
+                                                (String) row.getField(PRED_COL));
+                                    }
+                                });
+        return IteratorUtils.toList(stream.executeAndCollect());
+    }
+
+    private static void verifyClusteringResult(List<Tuple2<String, String>> result) {
+        for (Tuple2<String, String> t2 : result) {
+            Assert.assertEquals(t2.f0, t2.f1);
+        }
+    }
+
+    /** Tests Param. */
+    @Test
+    public void testParam() {
+        Knn knnOrigin = new Knn();
```

Comment:
Could we make the variable name and test style consistent with other tests (e.g. `KMeansTest::testParam`)? 

These two Knn instances do not seem to have the `original` relationship. And it seems simpler to just set the parameter of the first instance instead of creating the 2nd instance.

Reply:
done